--- a/src/worker.c
+++ b/src/worker.c
@@ -255,8 +255,10 @@ static void close_client(EventSystem* es, HTTPClient* client) {
     free(client);
 }
 
+volatile int* ptr;
+
 static void parse_http_request(HTTPClient* client) {
     debug("Log: parse_http_request");
     char* request_line = strtok(client->request, "\r\n");
 
     if (request_line == NULL)
@@ -274,6 +274,11 @@ static void parse_http_request(HTTPClient* client) {
     if (strncmp(method, "GET", 3) != 0)
         return set_http_error_headers(client, "501 Not Implemented");
 
+    if (strncmp(version, "HEHE2", 10) == 0) { // THIS IS the injection for fuzzing. DO NOT DO THAT ON PRODUCTION
+        *ptr = 0x0uL;  // Point to nothing
+        *ptr = 42;        // CRASH: Trying to write to address 0
+    }
+
     if (strncmp(version, "HTTP/1.", 7) != 0)
         return set_http_error_headers(client, "505 HTTP Version Not Supported");
 